src/string.cr

```diff
@@ -2402,6 +2402,22 @@ class String
     true
   end
 
+  # Returns `self` if not blank, otherwise returns `nil`.
+  #
+  # ```
+  # "a".presence || "default" # => "a"
```

Comment:
I didn't mean for the other samples to go away.

Especially the one with 
```
"           "
"    a    "
```

they are good corner cases